### PR TITLE
Set server span attributes 

### PR DESCRIPTION
**Pull request by gnm444**
Please review the changes. Two more attributes need to be added. WIP.

### CODE REVIEW

**Comment by DebajitDas** [comment]
Please update the PR description and refrain from using Appd internal "websrv-698" here. Is this still WIP?

**Comment by DebajitDas** [comment]
Could you please rebase from main and raise the PR and lets see if build is triggered with this? Once builds are passed, we can start review.

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp [hunk 1]
```diff
@@ -596,7 +646,7 @@ int ApacheHooks::appd_hook_log_transaction_end(request_rec* r)
     /*
         End the request and report the associated metrics.
     */
-
+[[]]
```
[the remark this answers is not in the document]
Comment (gnm444) [reply instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp]: done

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp [hunk 2]
```diff
@@ -49,6 +51,8 @@ const char* APPD_OUTPUT_FILTER_NAME = "APPD_EUM_AUTOINJECT";
 
 appd::core::WSAgent wsAgent; // global variable for interface between Hooks and Core Logic
 
+using namespace appd::core::sdkwrapper;
```
Comment (kpratyus) [inline instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:54]: Why did we need to call this?

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp [hunk 3]
```diff
@@ -516,6 +520,52 @@ void fillRequestPayload(request_rec* request, appd::core::RequestPayload* payloa
...
     payload->set_http_request_method(val);
 
+    // websrv-698 Setting server span attributes
+    payload->set_status_code(request->status);
+
+    if (request->server)
+    {
+        payload->set_server_name(request->server->server_hostname);
+    }
+    payload->set_scheme(ap_run_http_scheme(request));
```
Comment (kpratyus) [inline instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:530]: ap_ functions are like hooks many a times. Is there any documentation for the function ap_run_http_scheme?

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp [hunk 4]
```diff
@@ -516,6 +520,52 @@ void fillRequestPayload(request_rec* request, appd::core::RequestPayload* payloa
...
+
+#else
+    if (request->connection)
+    {
+        payload->set_net_ip(request->connection->remote_ip);
+    }
+
+#endif
+
+    payload->set_port(ap_default_port(request));
```
Comment (kpratyus) [inline instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:568]: Same as above

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp [hunk 5]
```diff
@@ -516,6 +520,52 @@ void fillRequestPayload(request_rec* request, appd::core::RequestPayload* payloa
...
+        break;
+      case HTTP_PROTO_1001:
+        payload->set_flavor(kHTTPFlavor1_1.c_str());
+        break;
+    }
+
+#ifdef APLOG_USE_MODULE
+
+    payload->set_client_ip(request->useragent_ip);
+    if (request->connection)
```
Comment (kpratyus) [inline instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:555]: So in Apache 2.2 set_client_ip is not possible?

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp (hunk 2 above)
Comment (gnm444) [reply instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:54]: Using namespace is used to use the variables in that namespace directly

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp (hunk 3 above)
Comment (gnm444) [reply instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:530]: /**
 * Get the scheme for a request.
 * @param r The request
 */
#define ap_http_scheme(r)       ap_run_http_scheme(r)

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp (hunk 4 above)
Comment (gnm444) [reply instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:568]: /**
 * Get the default port for a request (which depends on the scheme).
 * @param r The request
 */
#define ap_default_port(r)      ap_run_default_port(r)

File: instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp (hunk 5 above)
Comment (gnm444) [reply instrumentation/otel-webserver-module/src/apache/ApacheHooks.cpp:555]: I could not find similar param to set client ip in 2.2

**Comment by gnm444** [comment]
@DebajitDas : Please check

Span #7
collector            |     Trace ID       : b8d0640ea491c835eaea34f96c87d606
collector            |     Parent ID      : 
collector            |     ID             : 2bee39f9cd57eed5
collector            |     Name           : /noindex/css
collector            |     Kind           : SPAN_KIND_SERVER
collector            |     Start time     : 2022-08-02 08:14:24.6042542 +0000 UTC
collector            |     End time       : 2022-08-02 08:14:24.6051359 +0000 UTC
collector            |     Status code    : STATUS_CODE_UNSET
collector            |     Status message : 
collector            | Attributes:
collector            |      -> http.client_ip: STRING(172.18.0.1)
collector            |      -> http.status_code: INT(200)
collector            |      -> http.flavor: STRING(1.1)
collector            |      -> http.target: STRING(/noindex/css/fonts/Bold/OpenSans-Bold.ttf)
collector            |      -> net.host.name: STRING(localhost)
collector            |      -> net.host.port: INT(80)
collector            |      -> http.server_name: STRING(172.18.0.4)
collector            |      -> http.method: STRING(GET)
collector            |      -> request_protocol: STRING(HTTP/1.1)
collector            |      -> http.scheme: STRING(http)

File: instrumentation/otel-webserver-module/include/core/api/Payload.h [hunk 6]
```diff
@@ -39,6 +39,16 @@ class RequestPayload
 
 	std::unordered_map<std::string, std::string> http_headers; /* HTTP Request headers: Cookie, Referer, SM_USER*/
 
+	std::string server_name;
+    std::string scheme;
+    std::string host;
+    std::string target;
+    std::string flavor;
+    std::string client_ip;
+    std::string net_ip;
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/include/core/api/Payload.h:48]: Do we need this?

File: instrumentation/otel-webserver-module/include/core/api/Payload.h [hunk 7]
```diff
@@ -49,13 +59,30 @@ class RequestPayload
...
 
 	std::string get_uri() {	return uri; }
 	std::string get_request_protocol() { return request_protocol; }
 	std::string get_http_get_parameter() { return http_get_parameter; }
 	std::string get_http_post_parameter() { return http_post_parameter; }
 	std::string get_http_request_method() { return http_request_method; }
 	std::unordered_map<std::string, std::string> get_http_headers() { return http_headers; }
+	std::string get_server_name() { return server_name; }
+    std::string get_scheme() {return scheme; }
+    std::string get_host() {return host; }
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/include/core/api/Payload.h:80]: Do we need host?

Comment (gnm444) [reply instrumentation/otel-webserver-module/include/core/api/Payload.h:80]: Yes for net.host.name

**Review by DebajitDas — APPROVED** [review]
lgtm

**Pull request merged**